src/main/java/org/craftercms/engine/service/context/SiteContext.java

```diff
@@ -415,34 +434,51 @@ public void startGraphQLSchemaBuild() throws GraphQLBuildException {
     }
 
     public void destroy() throws CrafterException {
-        state = State.DESTROYED;
+        boolean locked;
+        try {
+            logger.debug("Getting write lock for context {}", this);
+            locked = writeLock.tryLock(shutdownTimeout, TimeUnit.MINUTES);
```

Comment:
This should be outside the `try`. `lock.lock()` are always put just before the `try`

Reply:
This methods can throw an `InterruptedException`, should we have 2 try blocks?

Reply:
Yes, there should be 2 levels of try, one which catches the `InterruptedException` and inside another try that has the finally that does the unlock. Also look at this example: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/Lock.html#tryLock(). They use `if`. In our case, we do the same whether we acquire the lock or not, but it's a good idea to do the if and log if the lock wasn't acquired and the timeout passed, so we can troubleshoot better.

Reply:
I'll add the second try, we have an if with debug messages is that enough?

Reply:
Yeah that's enough, I didn't check the debug line sorry